components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1654,6 +1655,68 @@ public OAuthRevocationResponseDTO revokeIssuedTokensByApplication(OAuthAppRevoca
         return revokeRespDTO;
     }
 
+    /**
+     * Revoke issued tokens for the application for the given authorized organization.
+     *
+     * @param application  {@link OAuthAppRevocationRequestDTO}.
+     * @param tenantDomain Tenant domain of the organization for which the tokens should be revoked.
+     * @return revokeRespDTO {@link OAuthAppRevocationRequestDTO}.
+     * @throws IdentityOAuthAdminException Error while revoking the issued tokens.
+     */
+    public OAuthRevocationResponseDTO revokeIssuedTokensForApplicationByOrganization
+    (OAuthAppRevocationRequestDTO application, String tenantDomain) throws IdentityOAuthAdminException {
+
+        triggerPreApplicationTokenRevokeListeners(application);
+        OAuthRevocationResponseDTO revokeRespDTO = new OAuthRevocationResponseDTO();
+
+        String consumerKey = application.getConsumerKey();
+        if (StringUtils.isBlank(consumerKey)) {
+            revokeRespDTO.setError(true);
+            revokeRespDTO.setErrorCode(OAuth2ErrorCodes.INVALID_REQUEST);
+            revokeRespDTO.setErrorMsg("Consumer key is null or empty.");
+            triggerPostApplicationTokenRevokeListeners(application, revokeRespDTO, new ArrayList<>());
+            return revokeRespDTO;
+        }
+
+        List<AccessTokenDO> accessTokenDOs = getActiveAccessTokensByConsumerKey(consumerKey);
+        if (!accessTokenDOs.isEmpty()) {
+            List<String> accessTokens = new ArrayList<>();
+            for (AccessTokenDO accessTokenDO : accessTokenDOs) {
+                String authorizedOrganizationId = accessTokenDO.getAuthorizedOrganizationId();
+                if (StringUtils.equals(getOrganizationId(tenantDomain), authorizedOrganizationId)) {
```

Comment:
getOrganizationId(tenantDomain) This will be a repetitive task. Lets fix this based on the method signature improvement.